File: apiEZPZ/src/main/java/com/example/apiezpz/rating/domain/Rating.java
package com.example.apiezpz.rating.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "ratings")
public class Rating {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "airport_id", nullable = false)
    private Airport airport;
    @Column(name = "airport_code", nullable = false)
    private String airportCode;

    @Column(nullable = false)
    private Integer satisfaction;
    @Column(nullable = false)
    private Integer cleanliness;
    @Column(nullable = false)
    private Integer convenience;

    @PrePersist
    public void prePersist() {
        if (this.airport != null) {
            this.airportCode = this.airport.getCode();
        }
    }
}

File: backend/src/main/kotlin/com/example/apiezpz/rating/domain/Rating.kt
package com.example.apiezpz.rating.domain

import jakarta.persistence.*

@Entity
@Table(name = "ratings")
data class Rating(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "airport_id", nullable = false)
    var airport: Airport? = null,
    
    @Column(name = "airport_code", nullable = false)
    var airportCode: String = "",

    @Column(nullable = false)
    var satisfaction: Int = 0,
    
    @Column(nullable = false)
    var cleanliness: Int = 0,
    
    @Column(nullable = false)
    var convenience: Int = 0
) {
    @PrePersist
    fun prePersist() {
        airport?.let {
            this.airportCode = it.code
        }
    }
} 